timelock-impl/src/main/java/com/palantir/atlasdb/timelock/ConjureResourceExceptionHandler.java

```diff
@@ -69,6 +69,16 @@ public ConjureResourceExceptionHandler(RedirectRetryTargeter redirectRetryTarget
                             // cancel operations. While this looks dodgy, it's intentional.
                             throw QosException.unavailable(interrupted);
                         },
+                        MoreExecutors.directExecutor())
+                .catching(
+                        RuntimeException.class,
+                        runtimeException -> {
+                            Throwable cause = runtimeException.getCause();
+                            if (cause instanceof InterruptedException) {
+                                throw QosException.unavailable(cause);
+                            }
+                            throw runtimeException;
```

Comment:
This makes sense. It is a bit worrying that we have to do this at the general level, but it's fine.